src/ResourceManager/DataLakeStore/Commands.DataLakeStore/DataPlaneModels/DataLakeStoreFileSystemClient.cs

```diff
@@ -45,8 +45,17 @@ public class DataLakeStoreFileSystemClient
         private const int MaxConnectionLimit = 1000;
         private const long NeverExpireValue = 253402300800000;
         internal const int ImportExportMaxThreads = 256;
+        private readonly LoggingConfiguration _adlsLoggerConfig;
+
         #region Constructors
 
+        static DataLakeStoreFileSystemClient()
+        {
+            // ConfigurationItemFactory.Default.Targets.RegisterDefinition("AdlsLogger", typeof(AdlsLoggerTarget));
+            Target.Register<Microsoft.Azure.Commands.DataLakeStore.Models.AdlsLoggerTarget>("AdlsLogger"); //generic
```

Comment:
@rahuldutta90 what is the purpose of this target in `NLog`?

Reply:
@cormacpayne NLog is used by the ADLS dataplane sdk to log debug messages. We can create a custom target which basically queues the debug data to the ConcurrentQueue (the one available to this powershell framework) for debug messages. So whenever we use our nlog framework to write debug message all of them will get queued in the concurrent queue. I have put some more comments in the target class.

Reply:
@cormacpayne  https://github.com/NLog/NLog/wiki/Extending-NLog